core/crypto/src/key_file.rs

```diff
@@ -22,16 +22,20 @@ pub struct KeyFile {
 
 impl KeyFile {
     pub fn write_to_file(&self, path: &Path) -> io::Result<()> {
-        let mut file = File::create(path)?;
-        #[cfg(unix)]
-        {
-            use std::os::unix::fs::PermissionsExt;
-            // Only the owner can read or write the file.
-            let perm = std::fs::Permissions::from_mode(0o600);
-            file.set_permissions(perm)?;
-        }
-        let str = serde_json::to_string_pretty(self)?;
-        file.write_all(str.as_bytes())
+        let mut file = Self::create(path)?;
+        let data = serde_json::to_string_pretty(self)?;
+        file.write_all(data.as_bytes())
+    }
+
+    #[cfg(unix)]
+    fn create(path: &Path) -> io::Result<File> {
+        use std::os::unix::fs::OpenOptionsExt;
+        std::fs::OpenOptions::new().mode(0o600).write(true).create(true).open(path)
```

Comment:
```suggestion
        std::fs::File::options().mode(0o600).write(true).create(true).open(path)
```

I think this is the preferred form nowdays.